lib/internal/child_process.js

```diff
@@ -269,6 +269,24 @@ function closePendingHandle(target) {
   target._pendingMessage = null;
 }
 
+function clearHandleQueue(target) {
+  const queue = target._handleQueue;
+  target._handleQueue = null;
+
+  const ex = new ERR_IPC_CHANNEL_CLOSED();
```

Comment:
Can you move this into the `if` block?  Callbacks and event listeners should get fresh exception objects so they don't step on each other when they modify it.